client/tests/integration/queries/role.rs

```diff
@@ -79,15 +80,16 @@ fn find_role_by_id() -> Result<()> {
     wait_for_genesis_committed(&vec![test_client.clone()], 0);
 
     let role_id: <Role as Identifiable>::Id = "root".parse().expect("Valid");
-    let role = NewRole::new(role_id.clone()).build();
+    let role = Role::new(role_id.clone());
 
     // Registering role
     let register_role = RegisterBox::new(role.clone());
     test_client.submit_blocking(register_role)?;
 
     let found_role = test_client.request(client::role::by_id(role_id))?;
 
-    assert_eq!(found_role, role);
+    assert_eq!(found_role.id(), role.id());
```

Comment:
Because `found_role` has type `Role` and `role` has type `NewRole`. I'm not sure if introducing `impl PartialEq<NewRole> for Role` would make much sense